tooling/ast_fuzzer/src/program/func.rs

```diff
@@ -729,4 +733,90 @@ impl<'a> FunctionContext<'a> {
 
         Ok(expr)
     }
+
+    /// Generate a `loop` loop.
+    fn gen_loop(&mut self, u: &mut Unstructured) -> arbitrary::Result<Expression> {
+        // Declare break index variable visible in the loop body.
+        let idx_type = types::U32;
+        let idx_id = self.next_local_id();
+        let idx_name = format!("idx_{}", make_name(idx_id.0 as usize, false));
+        let idx_ident =
+            expr::ident_inner(VariableId::Local(idx_id), true, idx_name.clone(), idx_type);
+
+        // Add a scope which will hold the index variable.
+        //self.locals.add(idx_id, false, idx_name.clone(), idx_type.clone());
+        self.locals.enter();
+
+        // Decrease budget so we don't nest endlessly.
+        self.decrease_budget(1);
+
+        // Initialize index to 0
+        let outer_block = Expression::Block(vec![Expression::Let(Let {
+            id: idx_id,
+            mutable: true,
+            name: idx_name,
+            expression: Box::new(expr::u32_literal(0)),
+        })]);
+
+        let inner_block = self.gen_block(u, &Type::Unit)?;
```

Comment:
If we use `gen_block` to generate the body of the loop, then I think the `self.locals.enter()` and `self.locals.exit()` isn't even needed, since we're not adding anything to the scope here and `gen_block` does it on its own.